### PR TITLE
React getting started improvements

### PR DESCRIPTION
**Pull request by GuillaumeSalles**
- Assume reader knows Typescript. 
- Introduce `createRoomProvider` generic types
- Defer introduction to `Suspense` to reduce the noise before we can show the first real time feature (who's in the room)

### CODE REVIEW

File: docs/pages/get-started/react.mdx [hunk 1]
```diff
@@ -9,6 +9,11 @@ In this guide, we’ll be learning how to add collaboration to React using the
 Liveblocks custom hooks. The hooks are part of [`@liveblocks/react`][], a
 package enabling multiplayer experiences in a matter of minutes.
 
+This guide uses [TypeScript](https://www.typescriptlang.org/). Liveblocks can
+definitely be used without TypeScript. We believe typings is helpful to make
```
Comment (nvie) [inline docs/pages/get-started/react.mdx:13]:
```suggestion
definitely be used without TypeScript. We believe typings are helpful to make
```

File: docs/pages/get-started/react.mdx [hunk 2]
```diff
@@ -62,7 +67,7 @@ Instead of using the client directly, we’re going to use [`createRoomContext`]
 from `@liveblocks/react` to create a [`RoomProvider`][] and hooks to make it
 easy to consume from our components.
 
-```jsx file="src/liveblocks.config.js" highlight="2,8-10"
+```tsx file="src/liveblocks.config.ts" highlight="2,8"
```
Comment (nvie) [inline docs/pages/get-started/react.mdx:70]:
```suggestion
```tsx file="liveblocks.config.ts" highlight="2,8"
```

**Review by nvie — APPROVED** [review]
LGTM! I like how you moved the suspense related stuff into a dedicated section! Glad that noise is now out of the primary part of the tutorial! I didn't double-check that all highlighted lines are correct this time. Have you?

**Review by CTNicholas — APPROVED** [review]
Good improvements, I like it! I've left a few suggestions that make it sound it a little more natural

**Comment by GuillaumeSalles** [comment]
Thank you both! I included all your recommendation and did a couple of fixes on the code snippets / highlighting

**Pull request merged**